src/core/PasswordGenerator.cpp

```diff
@@ -52,47 +53,99 @@ QString PasswordGenerator::generatePassword() const
 {
     Q_ASSERT(isValid());
 
-    const QVector<PasswordGroup> groups = passwordGroups();
+    const QVector<PasswordGroup> groups_original = passwordGroups();
+    QVector<PasswordGroup> groups = groups_original;
 
-    QVector<QChar> passwordChars;
+    QVector<QChar> passwordChars_original;
     for (const PasswordGroup& group : groups) {
         for (QChar ch : group) {
-            passwordChars.append(ch);
+            passwordChars_original.append(ch);
         }
     }
+    QVector<QChar> passwordChars = passwordChars_original;
 
     QString password;
+    //qDebug().noquote().nospace() << "*** Begin generating password ***";
 
     if (m_flags & CharFromEveryGroup) {
-        for (int i = 0; i < groups.size(); i++) {
-            int pos = randomGen()->randomUInt(groups[i].size());
-
-            password.append(groups[i][pos]);
-        }
 
-        for (int i = groups.size(); i < m_length; i++) {
-            int pos = randomGen()->randomUInt(passwordChars.size());
-
-            password.append(passwordChars[pos]);
+        /*
+         * Continuously add one random character from each group until pass-
+         * word has reached the desired length
+         *
+         * NOTE:
+         * Same as below, but more readable, just for reference
+         */
+        /*while (password.length() < m_length) {
```

Comment:
Ditto.

Reply:
Also don't use doc comments inside functions.

Reply:
This is just a multiline comment, doc comments start with '/**'.

Reply:
It's a hybrid. Don't start every line with an asterisk is what I wanted to say.